### DISCUSSION TITLE
RE: "If I write a misc (/dev/ge) driver for the #miyoomini will someone write patches for SDL1 to use it?"

### DISCUSSION THREAD

**Discussion by JohnnyonFlame** [linux-chenxing/linux-chenxing.org#57] [General]
If you add support for rotation via KMSDRM props you can always use our SDL1.2 fork :)
<https://github.com/OpenDingux/SDL/commits/fix-10bit>

**Comment by fifteenhex** [comment]
Thanks. That's interesting.

It looks like you have a KMSDRM driver that doesn't need libgbm, mesa and all that junk.. would it be a lot of work to make it work with SDL2? So we can have SDL2 + the SDL1 compat lib? I think then we can write a hardware accelerated renderer for SDL2 that uses the 2D graphics hardware.

**Reply by JohnnyonFlame** [reply]
Precisely, it's just a dumb buffer flipper w/ fully atomic kmsdrm using pure libdrm, with a few neat tricks :)
It's been tested far more throughout on OpenDingux, with a pretty wide range of applications, but not as much on other platforms, but it should work fine and we always accept patches.

> would it be a lot of work to make it work with SDL2? So we can have SDL2 + the SDL1 compat lib? I think then we can write a hardware accelerated renderer for SDL2 that uses the 2D graphics hardware.

I don't think it's a looooot of work, but I don't have the time for it right now, gotta clear up the backlog first before I can check lengthier projects.

Also I don't have a Miyoo Mini, my budget here isn't exactly big, I'm just following the developments from afar.

**Reply by fifteenhex** [reply]
Assuming aliexpress will ship to your country I don't mind getting you one in return for having a go at the SDL2 stuff.
I think we'll need that work to make other machines with the same SoC as the miyoo mini useful.

**Reply by JohnnyonFlame** [reply]
> Assuming aliexpress will ship to your country I don't mind getting you one in return for having a go at the SDL2 stuff. I think we'll need that work to make other machines with the same SoC as the miyoo mini useful.

I am flattered, but also am rather reserved about receiving straight up hardware samples from anyone but vendors for a few moral POVs:
- There's an implication of work being commissioned.
- It's not fair to burden foss contributors economically when they are already providing **very expensive** development hours such as you have.
- It's just a hobby in the end, and I don't want to pressure myself into pulling unreasonable hours, which I tend to do.

I am far more "laissez-faire" about straight donations, e.g. ko-fi, which are usually converted into hardware/software. But I'd rather explore other venues first.

I'm sorry for being pedantic, but fairness and transparency is the best way I have of making sure I respect you and other contributors.

**Reply by fifteenhex** [reply]
That's fair enough. I mainly didn't want to ask you to look into something you have no actual use for.
That would be pretty close to saying "hey, do this thing for me in your free time because I need it" which is pretty much my biggest annoyance with OSS.

**Comment by fifteenhex** [comment]
No longer needed. DirectFB2 handles the rotation now.
